core/src/executor/alter/table.rs

```diff
@@ -20,28 +29,85 @@ pub async fn create_table<T: GStore + GStoreMut>(
+                        name: source_name, ..
+                    } = &select_query.from.relation
+                    {
+                        let table_name = get_name(source_name)?;
+                        let schema = storage.fetch_schema(table_name).await?;
+                        let Schema {
+                            column_defs: source_column_defs,
+                            ..
+                        } = schema.ok_or_else(|| -> Error {
+                            AlterError::CtasSourceTableNotFound(table_name.to_owned()).into()
+                        })?;
+                        source_column_defs
+                    } else {
+                        return Err(Error::Table(TableError::Unreachable));
+                    }
                 }
-            }
-            _ => column_defs.to_vec(),
+                SetExpr::Values(Values(values_list)) => {
+                    let first_len = values_list[0].len();
+                    let init_types = iter::repeat(None)
+                        .take(first_len)
+                        .collect::<Vec<Option<DataType>>>();
+                    let column_types = values_list
+                        .iter()
+                        .fold_while(Ok(init_types), |column_types, exprs| {
+                            let result = column_types.and_then(
+                                |column_types| -> Result<(Vec<Option<DataType>>, bool)> {
+                                    let column_types = column_types
+                                        .iter()
+                                        .zip(exprs.iter())
+                                        .map(|(column_type, expr)| -> Result<_> {
+                                            let column_type = match column_type {
+                                                Some(data_type) => Some(data_type.to_owned()),
+                                                None => evaluate_stateless(None, expr)
+                                                    .and_then(Value::try_from)?
+                                                    .get_type(),
+                                            };
+
+                                            Ok(column_type)
+                                        })
+                                        .collect::<Result<Vec<Option<DataType>>>>()?;
+
+                                    let has_none = column_types.iter().any(Option::is_none);
+
+                                    Ok((column_types, has_none))
+                                },
+                            );
+
+                            match result {
+                                Ok((column_types, true)) => Continue(Ok(column_types)),
+                                Ok((column_types, false)) => Done(Ok(column_types)),
+                                Err(error) => Done(Err(error)),
+                            }
+                        })
+                        .into_inner();
```

Comment:
Rather than using `itertools`, we can use native iterator methods to solve this.
```rust
// by using
use std::ops::ControlFlow::{Break, Continue};

// e.g.
let column_types =
    values_list
        .iter()
        .try_fold(Ok(init_types), |column_types, exprs| {
            let result = column_types.and_then(
                |column_types| -> Result<(Vec<Option<DataType>>, bool)> {
                    let column_types = column_types
                        .iter()
                        .zip(exprs.iter())
                        .map(|(column_type, expr)| -> Result<_> {
                            let column_type = match column_type {
                                Some(data_type) => Some(data_type.to_owned()),
                                None => evaluate_stateless(None, expr)
                                    .and_then(Value::try_from)?
                                    .get_type(),
                            };

                            Ok(column_type)
                        })
                        .collect::<Result<Vec<Option<DataType>>>>()?;

                    let has_none = column_types.iter().any(Option::is_none);

                    Ok((column_types, has_none))
                },
            );

            match result {
                Ok((column_types, true)) => Continue(Ok(column_types)),
                Ok((column_types, false)) => Break(Ok(column_types)),
                Err(error) => Break(Err(error)),
            }
        });

let column_types = match column_types {
    Continue(column_types) | Break(column_types) => column_types,
};
```